ts/components/dialog/UserProfileModal.tsx

```diff
@@ -0,0 +1,217 @@
+  const profileName = useConversationUsername(conversationIdToDisplay) || '';
+  const [isEnlargedImageShown, setIsEnlargedImageShown] = useState(false);
+  const avatarSize = isEnlargedImageShown ? AvatarSize.HUGE : AvatarSize.XL;
+
+  function closeDialog() {
+    dispatch(updateUserProfileModal(null));
+  }
+
+  const [mode, setMode] = useState<Exclude<ProfileDialogModes, 'edit'>>('default');
+
+  const hasDisabledMsgRequests = useHasDisabledBlindedMsgRequests(conversationId);
+  const blindedAndDisabledMsgRequests = isBlindedAndNotResolved && hasDisabledMsgRequests;
+
+  async function onClickStartConversation() {
+    if (isBlindedAndNotResolved && hasDisabledMsgRequests) {
+      return;
+    }
+    const convo = ConvoHub.use().get(conversationIdToDisplay);
+
+    const conversation = await ConvoHub.use().getOrCreateAndWait(
+      convo.id,
+      ConversationTypeEnum.PRIVATE
+    );
+
+    await openConversationWithMessages({ conversationKey: conversation.id, messageId: null });
+
+    closeDialog();
+  }
+
+  useKey(
+    'Enter',
+    () => {
+      void onClickStartConversation();
+    },
+    undefined
+  );
+
+  return (
+    <SessionWrapperModal
+      headerChildren={<ModalBasicHeader title={''} showExitIcon={true} />}
+      onClose={closeDialog}
+      buttonChildren={
+        <ModalActionsContainer extraBottomMargin={true}>
+          <SessionButton
+            text={tr('message')}
+            onClick={onClickStartConversation}
+            buttonColor={SessionButtonColor.PrimaryDark}
+            dataTestId="new-session-conversation"
+            style={{ minWidth: '125px' }}
+            disabled={isBlindedAndNotResolved && hasDisabledMsgRequests}
+          />
+
+          {!isBlindedAndNotResolved && (
+            <CopyToClipboardButton
+              copyContent={conversationIdToDisplay}
+              buttonColor={SessionButtonColor.PrimaryDark}
+              dataTestId="copy-button-account-id"
+              style={{ minWidth: '125px' }}
+              hotkey={true}
+            />
+          )}
+        </ModalActionsContainer>
+      }
+    >
+      <Flex
+        $container={true}
+        width={'100%'}
+        $justifyContent="center"
+        $alignItems="center"
+        $flexDirection="column"
+        $flexGap="var(--margins-md)"
+        paddingBlock="0 var(--margins-lg)"
+      >
+        {mode === 'qr' ? (
+          <QRView sessionID={conversationIdToDisplay} setMode={setMode}>
+            <SessionLucideIconButton
+              unicode={LUCIDE_ICONS_UNICODE.USER_ROUND}
+              iconSize={'large'}
+              backgroundColor="var(--primary-color)"
+              iconColor="var(--black-color)"
+              padding="var(--margins-xs )"
+              onClick={() => {
+                setMode('default');
+              }}
+              style={{
+                position: 'absolute',
+                top: '-15px',
+                right: '-15px',
```

Comment:
not sure if we intend this to be on the other side in rtl mode or not? if so we can use inline-start

Reply:
have made it so